### PR TITLE
Add when bubblegum megafauna devours corpses it now spawns demonic bubblegum candy

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This spawns a demonic bubblegum candy if the Bubblegum megafauna successfully devours a corpse.

## Why It's Good For The Game

_♫ - Bubblegum bubblegum in a dish, how many corpses do you wish? - ♫_

## Changelog

:cl: timothymtorres
add: Bubblegum now spawns demonic bubblegum candy when they devour corpses.
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
Adds a dark twist on how nanotransen keeps their mining vendors stocked up.

Bubblegum gum is people!!!

**Comment by timothymtorres** [comment]
> Bubblegum gum is people!!!

Added it as a possible possible taste selection via https://github.com/tgstation/tgstation/pull/68701/commits/d18b5a2eb0eef12e4ad1aba2fb762217eff57c66

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 1]
```diff
@@ -228,7 +228,11 @@ Difficulty: Hard
 				addtimer(CALLBACK(src, .proc/devour, L), 2)
 	SLEEP_CHECK_DEATH(1, src)
 
-
+/mob/living/simple_animal/hostile/megafauna/bubblegum/devour(mob/living/yummy_food)
+	. = ..()
+	if(. == TRUE) // a corpse was devoured
+		// bubblegum bubblegum in a dish, how many corpses do you wish?
+		new /obj/item/food/bubblegum/bubblegum(src)
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:235]: You're spawning the bubblegum INSIDE the bubblegum mob, not on the floor where the body is killed.

Comment (timothymtorres) [reply code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:235]: Sounds like a feature. Buuuuut I guess we do want the bubblegum to be eaten by _human_ mobs and not bubblegum himself...

Comment (timothymtorres) [reply code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm]: Fixed with a58570e8753253ce89eee3dfe1df10086aee27b5

**Pull request merged**